pubsub.go

```diff
@@ -208,10 +262,26 @@ func (p *PubsubValueStore) handleSubscription(sub *floodsub.Subscription, key st
 			if err != nil {
 				log.Warningf("PubsubResolve: error writing update for %s: %s", key, err)
 			}
+			p.notifyWatchers(key, msg.GetData())
 		}
 	}
 }
 
+func (p *PubsubValueStore) notifyWatchers(key string, data []byte) {
+	p.watchLk.RLock()
+	sg, ok := p.watching[key]
+	p.watchLk.RUnlock()
+	if !ok {
+		return
+	}
+
+	sg.lk.RLock()
+	for _, watcher := range sg.listeners {
+		watcher <- data
```

Comment:
This *also* (in addition to the context) needs to select on `sg.closing` as we're not holding the watchLk. Currently, `sg.closing` can be closed, making all the listeners exit, and we'll hang here forever.